src/client/opamCommands.ml

```diff
@@ -2580,6 +2582,20 @@ let repository cli =
            '--all' to see all configured repositories.";
       OpamRepositoryCommand.list rt ~global ~switches ~short;
       `Ok ()
+    | Some `show_repo, [ repo_name ] ->
+      OpamRepositoryState.with_ `Lock_none gt @@ fun rt ->
+      let switches =
+        if scope = [] ||
+           List.exists (function
+               | `This_switch | `Current_switch | `Switch _ -> true
+               | _ -> false)
```

Comment:
Could you list all the variants explicitly? It help for detecting required changes for future maintenance.
I also think that unsupported options should probably display an error message.